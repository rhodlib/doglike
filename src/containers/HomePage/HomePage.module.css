.HomePage {
  padding: 0 1rem;
}

.Title {
  margin: 0;
  padding: 2rem 0 2rem 0.5rem;
}

.Header {
  display: flex;
  margin-bottom: 2rem;
  justify-content: space-between;
  flex-wrap: wrap;
  align-items: center;
}