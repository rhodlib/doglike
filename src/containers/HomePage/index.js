import React, { useEffect, useState } from "react";
import Title from "../../components/Title";
import CardList from "../../components/CardList";
import Footer from "../../components/Footer";
import { getDogArray } from "../../core/actions/racesActions";
import { useDispatch, useSelector } from "react-redux";
import styles from "./HomePage.module.css";
import SearchBar from "../../components/SearchBar";

/**
 * Component that shows the Home and receives an array.
 */

export const HomePage = () => {
  const dispatch = useDispatch();
  const dogResponse = useSelector(state => state.raceReducer);
  const [filter, setFilter] = useState("");

  // Getting array of dog breeds from store.
  useEffect(() => {
    dispatch(getDogArray());
  }, [dispatch]);

  // Return a fragment with a title, list of cards and a footer.
  return (
    <main className={styles.HomePage}>
      <header className={styles.Header}>
        <Title name={"dog like"} className={styles.Title} />
        <SearchBar
          placeholder="Search dog race"
          dogList={dogResponse.races}
          defaultValue={filter}
          onInput={event => setFilter(event.currentTarget.value)}
        />
      </header>
      <CardList
        dogList={(dogResponse.races || []).filter(race =>
          race.includes(filter)
        )}
      />
      <Footer />
    </main>
  );
};

export default HomePage;
