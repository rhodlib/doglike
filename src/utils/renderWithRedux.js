import React from "react";
import { Provider } from "react-redux";
import { createStore, applyMiddleware } from "redux";
import reducers from "../core/reducers";
import { initialState as defaultInitialState } from "../core";
import { render } from "@testing-library/react";
import reduxThunk from "redux-thunk";

/**
 * This is a Util to render test with redux.
 */
export const renderWithRedux = (
  ui,
  {
    initialState = defaultInitialState,
    store = createStore(reducers, initialState, applyMiddleware(reduxThunk))
  } = {}
) => ({ ...render(<Provider store={store}>{ui}</Provider>), store });

export default renderWithRedux;
