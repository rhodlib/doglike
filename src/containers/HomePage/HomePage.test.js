import React from "react";
import HomePage from "./index";
import renderWithRedux from "../../utils/renderWithRedux";

it("Matches snapshot", () => {
  const { asFragment } = renderWithRedux(<HomePage />);
  expect(asFragment()).toMatchSnapshot();
});
