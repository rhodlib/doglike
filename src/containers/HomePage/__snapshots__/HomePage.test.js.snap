// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <main
    class="HomePage"
  >
    <header
      class="Header"
    >
      <h1
        class="Title Title"
      >
        dog like
      </h1>
      <input
        class="SearchBar"
        list="dogs"
        placeholder="Search dog race"
        value=""
      />
      <datalist
        id="dogs"
      />
    </header>
    <div
      class="CardList"
    />
    <div
      class="Footer"
    >
      <p
        class="FooterParagraph"
      >
        Created by 
        <a
          class="FooterAnchor"
          href="https://rhodlib.me"
          rel="noopener noreferrer"
          target="_blank"
        >
          Rhodlib
        </a>
      </p>
    </div>
  </main>
</DocumentFragment>
`;
